goclone

```diff
@@ -147,13 +163,24 @@ if [[ ! -d "${clone_path}" ]]; then
     fi
 fi
 
+if [[ ! -z "${branch_name:-}" ]]; then
+    if [[ ! -d "${branch_project}" ]]; then
+        (
+            cd "${clone_path}"
+            git worktree add "${branch_clone_path}" "${branch_name}"
```

Comment:
I don't know what this does, why is it necessary? Can't you just clone and simply add a git checkout next to the clone?

Reply:
`git worktree` in general or?

My intent for doing this wasn't just to save an extra `git checkout` but was to sort of automate the worktree process for keeping feature branches off of a single clone. For example, the kubernetes repo is large enough that I really don't want to have to fetch multiple ties.

This will make a full GOPATH (so branches can diverge from one another with regards to imports/versions) and so builds/etc work (tons of things break, for example, if I just cloned/worktreed into ../kubernetes2 since GOPATH would be wrong, etc, etc).

Reply:
Sounds good, I'm not super familiar but I guess it's better than a hard-clone, but I assumed you weren't worried about wasted disk space.